src/modules/household/components/elements/AddToHouseholdButton.tsx

```diff
@@ -46,25 +54,73 @@ const AddToHouseholdButton = ({
   if (added) text = 'Added';
   const clientId = client.id;
 
+  const [conflictingEnrollmentId, setConflictingEnrollmentId] = useState<
+    string | undefined
+  >();
+
   const memoedArgs = useMemo(
     () => ({
       formRole: RecordFormRole.Enrollment,
       onCompleted: (data: SubmittedEnrollmentResultFieldsFragment) => {
         setAdded(true);
         onSuccess(data.householdId);
       },
-      inputVariables: { projectId, clientId },
-      pickListArgs: { projectId, householdId },
+      inputVariables: { projectId: project.id, clientId },
+      pickListArgs: { projectId: project.id, householdId },
       localConstants: { householdId, projectCocCount: cocCount },
+      errorFilter: (error: ValidationError) => {
+        // If there's an error about a conflicting enrollment, we will show the ConflictingEnrollmentAlert,
+        // so here, we filter that error out of the ErrorAlert displayed by the form dialog.
+        if (!householdId) return true; // Except if this is a new household. Then show the error, since we can't Join Households.
+        if (!error.data) return true;
+        return !error.data?.hasOwnProperty('conflictingEnrollmentId');
```

Comment:
maybe slightly easier to follow?
```suggestion
        // If there's an error about a conflicting enrollment, and we're adding to an existing household,
        // then we will show the ConflictingEnrollmentAlert (so filter it out from the ErrorAlert display)
        if (householdId && error.data?.hasOwnProperty('conflictingEnrollmentId')) {
          return false;
        }
        return true;
```

Reply:
Actually it looks like we only show the error if `household` is present,  so we should be checking that instead of `householdId`.
Can you clarify the need for having both props `householdId` and `household`? Are there time when this component receives only the ID but not the household record, and what are the implications?

Reply:
Hm, let me try to think it through. The existing code flow is something like this:
- `ManageHousehold` receives the `householdId` from the [`AllEnrollmentDetailsFragment` on the Enrollment dashboard](https://github.com/greenriver/hmis-frontend/blob/release-151/src/modules/household/components/EditHouseholdPage.tsx#L11), meaning that the _household_ itself hasn't been fetched and isn't in memory yet, only the ID.
- `ManageHousehold` then passes that "initial" `householdId` along to the [`useAddToHouseholdColumns` hook](https://github.com/greenriver/hmis-frontend/blob/release-151/src/modules/household/hooks/useAddToHouseholdColumns.tsx)
- `useAddToHouseholdColumns`
  - provides a function for fetching the household;
  - and separately, it provides a renderer for the [`AddToHouseholdButton`, passing it the `householdId`](https://github.com/greenriver/hmis-frontend/blob/release-151/src/modules/household/hooks/useAddToHouseholdColumns.tsx#L79) (not the whole `household`)

Changes introduced in this PR:
- `AddToHouseholdButton` now is the parent of the whole Join workflow, so it needs the whole `household`
- We already have this in `data.household` in `useAddToHouseholdColumns` so I added it as a new prop to `AddToHouseholdButton`.

So, the short answer to your original question,
> Can you clarify the need for having both props householdId and household?

is that I was trying to update the component to get the data about `household` it now needs, without messing with the existing behavior that depends on `householdId`, for fear of introducing new bugs.

However, [`ManageHousehold`](https://github.com/greenriver/hmis-frontend/blob/a028a949d9012265cd12393602bf0864d1b2466e/src/modules/household/components/ManageHousehold.tsx#L129) (which is ultimately the renderer of the `AddToHouseholdButton` returned by `useAddToHouseholdColumns` via `addToEnrollmentColumns` ... whew) returns `Loading` if `household` is undefined but an initial household ID was provided. So, _maybe_, we can rely on the idea that `householdId` and `household` will always match by the time `addToEnrollmentColumns` gets rendered? And thereby get rid of the `householdId` prop on the `AddToHouseholdButton`?

Reply:
Ok, sorry it's a bit of a tangle. I agree it would be preferable to remove the `householdId` prop and only use `household`. 

It think the only time this might cause a blip is in the following scenario:
* Project => Add Enrollment
* Search => "Enroll Client"
  * After the modal closes, the buttons automatically update to "Add to Household" text and have the `householdId` in state. Whereas if we're waiting for the household to fetch (which gets triggered in `useAddToHouseholdColumns` when the householdId changes), we need to wait another moment until those buttons receive the `household` state.
  * This could be addressed by passing a disabled state to `AddToHouseholdButton` where `disabled={loading && !!householdId && !data?.household}`, so those buttons are disabled in those seconds where the household is still loading. (Purpose being to prevent another "enroll client" mutation that spins up a _new_ household).
  * (Also keeping in mind the new enrollment designs have us unmounting the search altogether after an enrollment is added. Taking a peek at the code, that might not be quite as straightforward as it sounds -- otherwise I would suggest including it here.)

Let me know if you'd like to pair on this

Reply:
Thanks Gig, that's helpful! I was worried about introducing bugs but I played around with it in the browser with the debugger to slow things down, and I think I feel confident in the behavior now.

I've implemented it as you suggested, but I was also wondering, do you think there is any harm in just disabling the button when `loading` is true (regardless of other conditions)? Maybe that logic would be easier to reason thru when we revisit this code in the future. The downside is that the user can't add multiple enrollments to a household as quickly, since it forces you to wait until the household refetches before you can click the "Add to Household" button again.

Reply:
Ok, great! Thank you! I think fewer disabled states is probably more user-friendly because there is less visually changing in the UI for the user to track/interpret.